backend/api/auth.py

```
@@ -0,0 +1,455 @@
+        except Exception as e:
+            logger.error(f"Password verification failed: {e}")
+            return False
+    
+    @staticmethod
+    def get_password_hash(password: str) -> str:
+        """Hash password"""
+        try:
+            return pwd_context.hash(password)
+        except Exception as e:
+            logger.error(f"Password hashing failed: {e}")
+            raise AuthenticationError("Failed to hash password")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)